### ISSUE TITLE
Background building of project solution

### ISSUE THREAD

**Issue by martinruefenacht** [godotengine/godot-proposals#849]
**Describe the project you are working on:**
I am working on a management based game with the frontend written in GDscript, but the backend written in C#.

This applies to any C# utilising project.

**Describe the problem or limitation you are having in your project:**
Currently this is not a huge issue, but the build time could start getting very annoying since the building forces a panel "Building project solution". This is essentially a useless panel and blocks any kind of work in the editor.

<img width="547" alt="screen" src="https://user-images.githubusercontent.com/825164/82142276-0271fc00-9833-11ea-8760-06916edcfdb2.png">

This is related to #241. These panels are (in my opinion) useless and distracting.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Move the building of the project into a background process and show the status (if required at all) to the mono builds panel with a progress bar, such that the editor can be used as normal.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Simply remove the project building panel or have a non-interrupting method of displaying the build progress. The same as #241.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

No.

**Is there a reason why this should be core and not an add-on in the asset library?:**

I don't think this is accessible by a plug in.

**Comment by Jummit** [comment]
The same could be said about baking lighting. Although I don't know if the editor would be usable even if you could interact with it whilst baking.

**Comment by salahchafai** [comment]
i agree with this so much, these panels should be reworked and have a different behviour, for this particular one i think it should be a background process just as the op said, but for the build failed panel i think it should be replaced by a red fading tooltip

**Comment by Calinou** [comment]
Related to #323 and #35.

**Comment by martinruefenacht** [comment]
@Calinou My current build time is 3 seconds, and that is a the beginning of this project.

I think just having a toast to give a progress bar would be fine in my opinion. My main concern is that it blocks using the editor during that time.

**Comment by neikeq** [comment]
Note, you can already build the solution in the background outside Godot (from command line or an IDE).
I plan to add background building in the Godot editor, but only when building manually. When launching the game, it will still be a blocking build, as that what the editor build hook API allows.
